lib/dry/types/extensions/monads.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'dry/monads/result'
+
+module Dry
+  module Types
+    # Monad extension for Result
+    #
+    # @api public
+    class Result
+      include Dry::Monads::Result::Mixin
+
+      # Turn result into a monad
+      #
+      # This makes result objects work with dry-monads (or anything with a compatible interface)
+      #
+      # @return [Dry::Monads::Success,Dry::Monads::Failure]
+      #
+      # @api public
+      def to_monad
+        if success?
+          Success(self)
```

Comment:
How do you feel about folding the Dry::Types::Result into the Dry::Monads::Result instead of nesting?

It would make Dry::Monads::Success<Dry::Types::Failure<x>> and Dry::Monads::Failure<Dry::Types::Success<x>> cases that a user doesn't have to think about for exhaustivity despite being impossible via this code path, and also just removes a level of nesting.

Reply:
though I see the nesting is exactly how it's done in schema, so I suppose my proposal is larger in scope than I intended.